tap_s3_csv/sync.py

```diff
@@ -70,9 +75,78 @@ def sync_table_file(config, s3_path, table_spec, stream):
         rec = {**row, **custom_columns}
 
         with Transformer() as transformer:
-            to_write = transformer.transform(rec, stream['schema'], metadata.to_map(stream['metadata']))
+            to_write = transformer.transform(
+                rec, stream['schema'], metadata.to_map(stream['metadata']))
 
         singer.write_record(table_name, to_write)
         records_synced += 1
 
     return records_synced
+
+
+def sync_jsonl_file(config, s3_path, table_spec, stream):
+    LOGGER.info('Syncing file "%s".', s3_path)
+
+    bucket = config['bucket']
+    table_name = table_spec['table_name']
+
+    file_handle = s3.get_file_handle(config, s3_path)._raw_stream
+    iterator = file_handle
+
+    records_synced = 0
+
+    for row in iterator:
+
+        decoded_row = row.decode('utf-8')
+        if decoded_row.strip():
+            row = json.loads(decoded_row)
+        else:
+            continue
+
+        custom_columns = {
+            s3.SDC_SOURCE_BUCKET_COLUMN: bucket,
+            s3.SDC_SOURCE_FILE_COLUMN: s3_path,
+
+            # index zero and then starting from 1
+            s3.SDC_SOURCE_LINENO_COLUMN: records_synced + 1
+        }
+        rec = {**row, **custom_columns}
+
+        with Transformer() as transformer:
+            to_write = transformer.transform(
+                rec, stream['schema'], metadata.to_map(stream['metadata']))
+
+        value = [{field: rec[field]} for field in set(rec) - set(to_write)]
```

Comment:
Could you add a comment to describe what this is doing? Its doing a set difference to check for what needs to be placed in the extra column?

Reply:
Yes, collecting the value which was removed in transform to add those in _sdc_extra. The comment also added.